mne/filter.py

```diff
@@ -2018,7 +2018,9 @@ def resample(self, sfreq, npad='auto', window='boxcar', n_jobs=1,
                  pad='edge', verbose=None):  # lgtm
         """Resample data.
 
-        If appropriate, an anti-aliasing filter is applied before resampling.
+        Through use of an FFT-based (fast Fourier transform) method for
+        resampling, aliasing artifacts are prevented as if an anti-aliasing
+        filter were applied beforehand.
```

Comment:
I would revert this. A brick-wall frequency-domain filter is an anti-aliasing filter. You can think of it as being equivalent to a time-domain (circular) convolution with a sinc if you want.

Reply:
cc @cbrnr you want to argue against this?

Else, I'll do as Eric says and go with my previous suggestion

Reply:
I guess the important thing is to let people know if they need to manually apply an anti-aliasing filter or not. Typically, most people think of a lowpass filter to do that, in this is not necessary here. I think I didn't like the previous sentence because you said that "an anti-aliasing filter is applied *before* resampling" - I would argue that it's not applied before, because it is part of the FFT/IFFT procedure.

Reply:
How about:

```suggestion
        To achieve anti-aliasing, this function applies a brick-wall
        frequency-domain filter during the FFT-based
        (fast Fourier transform) resampling.
```

Reply:
> So technically the statement that MNE adds an anti-aliasing filter is not correct because this is not a separate step.

Quoting the link posted earlier:

> Alternatively, you can see your approach as filtering with an aliasing filter (i.e. lowpass with cutoff 1/M) before downsampling.

The fact that it's not manually/separately applied *in the code* is an implementation detail. It's mathematically equivalent to doing it, so we should let people sleep better at night by saying that an anti-aliasing filter is applied to their data. To me this is similar to not needing to tell people that we do overlap-add frequency-domain convolution when we filter, we can just say that the output is the convolution of the signal with the FIR filter because it's equivalent. (And FWIW actually some years ago we didn't use overlap-add or direct convolution but instead a one-shot full frequency-domain convolution method, too. But people didn't need to care when we switched, and we only bothered documenting it because it was a speed improvement.)

There are things we can and probably should discuss somewhere about these methods -- such as the extent to which using a circular convolution with a sinc / freq-domain brick-wall is risky -- but I wouldn't equivocate about whether or not an anti-aliasing filter is applied. The natural question when reading the new text is, "so wait, no anti-aliasing filter is applied? So there can be aliasing?" It's just going to needlessly make people worry.

Reply:
+1 for the brick-wall language, it's explicit, technically correct, and should let people sleep :)

Reply:
(though technically the `window` argument allows things other than a strict brick-wall filter -- the default `'boxcar'` just so happens to be the brick wall)

Reply:
I agree that these are really technical details at the algorithm level, but then at least remove the word "before" to let me sleep better at night :smile:.

Also, when people go inspect the code and don't find the anti-aliasing filter (because they are expecting that some filter is applied before resampling), it would be good to discuss the FFT-based method somewhere to explain that it incorporates an anti-aliasing filter. I'd be interested in this, e.g. I haven't bothered to get into the details such as how do we handle cases where signal lengths are prime (do we accept that resampling will be slow then, or would it be better to switch to a polyphase implementation you mentioned before), and so on.

Anyway, I let you decide on the wording @larsoner.